implementations/rust/ockam/ockam_api/src/nodes/iolets.rs

```diff
@@ -6,40 +6,106 @@ use ockam_core::compat::borrow::Cow;
 #[cfg(feature = "tag")]
 use ockam_core::TypeTag;
 
-/// Request body to create an inlet
-#[derive(Debug, Clone, Decode, Encode)]
+/// Distinguish between inlet and outlet portals in the API
+#[derive(Copy, Clone, Debug, Decode, Encode, PartialEq)]
+#[rustfmt::skip]
+#[cbor(index_only)]
+pub enum IoletType {
+    #[n(0)] Inlet,
+    #[n(1)] Outlet,
+}
+
+/// Request body to create an inlet or outlet
+#[derive(Clone, Debug, Decode, Encode)]
 #[rustfmt::skip]
 #[cbor(map)]
-pub struct CreateInlet<'a> {
+pub struct CreateIolet<'a> {
     #[cfg(feature = "tag")]
     #[n(0)]
     tag: TypeTag<1407961>,
-    #[n(1)] pub addr: Cow<'a, str>,
+    /// The type of portal endpoint to create
+    #[n(1)] pub tt: IoletType,
+    /// The address the portal should connect or bind to
+    #[n(2)] pub addr: Cow<'a, str>,
```

Comment:
Shouldn't you use `#[b(2)]` ?

Reply:
Oh yea probably. I've just been using `#[n(_)]` for all the fields at the moment